services/checks/registry.go

```diff
@@ -0,0 +1,76 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package checks
+
+import (
+	"sync"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/percona/pmm/api/alertmanager/ammodels"
+)
+
+// registry stores alerts and delay information by IDs.
+type registry struct {
+	rw     sync.RWMutex
+	alerts map[string]ammodels.PostableAlert
```

Comment:
why do we still use map here? Looks like it's unnecessary anymore

Reply:
right, will change it to a slice in the next push

Reply:
I guess we can rename `alertWithID` to just `alert` (we don't use that name yet, right?) and store `alerts []alert`.